bin/kamstrup_prober.py

```diff
@@ -93,74 +93,88 @@ def get_register(self, register):
-parser.add_argument("port", type=int, help="TCP port")
-parser.add_argument(
-    "--registerfile",
-    dest="registerfile",
-    help="Reads registers from previous dumps files instead of"
-    "bruteforcing the meter.",
-)
-parser.add_argument("--comaddress", dest="communication_address", default=0x3F)
-args = parser.parse_args()
-
-if args.registerfile:
-    candidate_registers_values = []
-    with open(args.registerfile, "r") as register_file:
-        old_register_values = json.load(register_file)
-    for value in old_register_values.iterkeys():
-        candidate_registers_values.append(int(value))
-else:
-    candidate_registers_values = range(0x01, 0xFFFF)
-
-kamstrupRegisterCopier = KamstrupRegisterCopier(
-    args.host, args.port, int(args.communication_address)
-)
-found_registers = {}
-not_found_counts = 0
-scanned = 0
-dumpfile = "kamstrup_dump_{0}.json".format(
-    calendar.timegm(datetime.utcnow().utctimetuple())
-)
-
-for x in candidate_registers_values:
-    result = kamstrupRegisterCopier.get_register(x)
-    if len(result) > 12:
-        units = result[5]
-        length = result[6]
-        unknown = result[7]
-
-        register_value = 0
-        for p in range(length):
-            register_value += result[8 + p] << (8 * ((length - p) - 1))
-
-        found_registers[x] = {
-            "timestamp": datetime.utcnow(),
-            "units": units,
-            "value": register_value,
-            "value_length": length,
-            "unknown": unknown,
-        }
-        print("Found register value at {0}:{1}".format(hex(x), register_value))
-        with open(dumpfile, "w") as json_file:
-            json_file.write(json.dumps(found_registers, indent=4, default=json_default))
-    else:
-        not_found_counts += 1
-        if not_found_counts % 10 == 0:
-            print(
-                "Hang on, still scanning, so far scanned {0} and found {1} registers".format(
-                    scanned, len(found_registers)
+    print(
+        "Scanned {0} registers, found {1}.".format(
+            len(candidate_registers_values), len(found_registers)
+        )
+    )
+    # with open('kamstrup_dump_{0}.json'.format(calendar.timegm(datetime.utcnow().utctimetuple())), 'w') as json_file:
+    #    json_file.write(json.dumps(found_registers, indent=4, default=json_default))
+    print("""*** Sample Conpot configuration from this scrape:""")
+    print(generate_conpot_config(found_registers))
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Probes kamstrup_meter meter registers.")
+    parser.add_argument("host", help="Hostname or IP or Kamstrup meter")
+    parser.add_argument("port", type=int, help="TCP port")
+    parser.add_argument(
+        "--registerfile",
+        dest="registerfile",
+        help="Reads registers from previous dumps files instead of"
+             "bruteforcing the meter.",
+    )
+    parser.add_argument("--comaddress", dest="communication_address", default=0x3F)
```

Comment:
make this an integer and the default a module level variable

Reply:
I'm pretty sure that this is the default port of 63. Please correct me if I'm wrong.

`default_comm_port = 63`

Reply:
thats correct :+1: